src/components/switch/switch.tsx

```diff
@@ -0,0 +1,37 @@
+import { BaseComponent } from "../common/base-component";
+import { Component, Event, EventEmitter, Prop } from "@stencil/core";
+import { SwitchRender } from "../renders/bootstrap/switch/switch-render";
+@Component({
+  host: {
+    role: "switch"
+  },
+  shadow: false,
+  styleUrl: "switch.scss",
+  tag: "gx-switch"
+})
+export class Switch extends SwitchRender(BaseComponent) {
+  /**
+   * Attribute that provides the caption to the <label> tag.
+   */
+  @Prop() caption: string;
+  /**
+   * Indicates that the checked is selected by default.
```

Comment:
I'd change this for something like:

Indicates if the switch is checked by default.